packages/react-meteor-data/react-meteor-data.d.ts

```diff
@@ -26,3 +26,7 @@ export function useSubscribe(name?: string, ...args: any[]): () => boolean;
 
 export function useFind<T>(factory: () => Mongo.Cursor<T>, deps?: React.DependencyList): T[];
 export function useFind<T>(factory: () => Mongo.Cursor<T> | undefined | null, deps?: React.DependencyList): T[] | null;
+
+export function usePromise<FN extends <T>(...args: any) => T & Promise<T>>
+(promise: (...inputs: Parameters<FN>) => Promise<ReturnType<FN>>,
+ inputs: Parameters<FN>, lifespan: number): Awaited<ReturnType<FN>>
```

Comment:
I don't think we should make it a part of our public API and rather keep it as an implementation detail. If needed, you can mark it as `@internal` in its JSDoc so TypeScript won't generate types for it.